### PR TITLE
[Core] Proposal to improve validate settings

### PR DESCRIPTION
**Pull request by ddiezrod**
This way you make sure all default settings from sub-levels are added.  I know this has been subject of debate, what do you think?

### CODE REVIEW

File: kratos/python_scripts/python_solver.py [hunk 1]
```diff
@@ -53,7 +53,8 @@ def ValidateSettings(self):
         """This function validates the settings of the solver
         """
         default_settings = self.GetDefaultSettings()
-        self.settings.ValidateAndAssignDefaults(default_settings)
+        self.settings.ValidateDefaults(default_settings)
```
Comment (loumalouomega) [inline kratos/python_scripts/python_solver.py:56]: Huuuuum, I need to think about this

Comment (philbucher) [reply kratos/python_scripts/python_solver.py:56]: same

**Comment by philbucher** [comment]
hm I don't think is possible
Usually the linear solver settings are the issue
Imaging the default settings to be like this:
~~~js
{ // solver settings
    ... // some solver settings
    "linear_solver_settings" : {
        "solver_type": "amgcl"
        ... // other amgcl parameters (INCOMPATIBLE with other solvers)
    }
}
~~~

now if you add the missing parameters, then the incompatible parameters will also be added and the validation in other linear solvers will fail

Thats also one of the reasons why we don't do recursive validation

I am not 100% this is a problem, but quite sure. You could give it a try...

**Comment by ddiezrod** [comment]
> hm I don't think is possible
> Usually the linear solver settings are the issue
> Imaging the default settings to be like this:
> 
> ```js
> { // solver settings
>     ... // some solver settings
>     "linear_solver_settings" : {
>         "solver_type": "amgcl"
>         ... // other amgcl parameters (INCOMPATIBLE with other solvers)
>     }
> }
> ```
> 
> now if you add the missing parameters, then the incompatible parameters will also be added and the validation in other linear solvers will fail
> 
> Thats also one of the reasons why we don't do recursive validation
> 
> I am not 100% this is a problem, but quite sure. You could give it a try...

I think you are right as usual :(  I'll keep thinking...

**Comment by rubenzorrilla** [comment]
Agree with @philbucher comments. Can we delete the branch?

**Comment by philbucher** [comment]
@ddiezrod as I wrote, I am not sure, you could try...

**Pull request closed without merging**